src/ext/java/org/opentripplanner/ext/fares/impl/DefaultFareService.java

```diff
@@ -102,18 +104,83 @@ public ItineraryFares calculateFares(Itinerary itinerary) {
     if (fareLegs.isEmpty()) {
       return null;
     }
+    var fareLegsByFeed = fareLegs
+      .stream()
+      .collect(Collectors.groupingBy(leg -> leg.getAgency().getId().getFeedId()));
+    var fareRulesByTypeAndFeed = fareRulesPerType
+      .entrySet()
+      .stream()
+      .collect(
+        Collectors.toMap(
+          Map.Entry::getKey,
+          rules ->
+            rules
+              .getValue()
+              .stream()
+              .collect(Collectors.groupingBy(rule -> rule.getFareAttribute().getId().getFeedId()))
+        )
+      );
 
     ItineraryFares fare = ItineraryFares.empty();
     boolean hasFare = false;
-    for (Map.Entry<FareType, Collection<FareRuleSet>> kv : fareRulesPerType.entrySet()) {
-      FareType fareType = kv.getKey();
-      Collection<FareRuleSet> fareRules = kv.getValue();
-      // Get the currency from the first fareAttribute, assuming that all tickets use the same currency.
-      if (fareRules.size() > 0) {
-        Currency currency = Currency.getInstance(
-          fareRules.iterator().next().getFareAttribute().getCurrencyType()
+    for (FareType fareType : fareRulesPerType.keySet()) {
+      List<FareComponent> components = new ArrayList<>();
+      List<Money> fares = new ArrayList<>();
+      ItineraryFares currentFare = ItineraryFares.empty();
+      boolean legWithoutRulesFound = false;
+      boolean legsWithoutMatchingRulesFound = false;
+      boolean fareTypeHasFare = false;
+      for (String feedId : fareLegsByFeed.keySet()) {
+        var fareRules = fareRulesByTypeAndFeed.get(fareType).get(feedId);
+
+        // Get the currency from the first fareAttribute, assuming that all tickets use the same currency.
+        if (fareRules != null && fareRules.size() > 0) {
+          Currency currency = Currency.getInstance(
+            fareRules.iterator().next().getFareAttribute().getCurrencyType()
+          );
+          boolean feedHasFare = false;
+          feedHasFare =
+            populateFare(currentFare, currency, fareType, fareLegsByFeed.get(feedId), fareRules);
+
+          if (!feedHasFare) legsWithoutMatchingRulesFound = true;
```

Comment:
```suggestion
          if (!feedHasFare) {
            legsWithoutMatchingRulesFound = true;
          }
```

Reply:
I wonder if prettier can be configured to insert these brackets. I took a quick look but couldn't find something. Do you know?

Reply:
I've wondered the same but haven't looked into it. I really find it difficult to read the code especially if value of a variable changes in the same line where there is some if statement.